Rework Migrate section into Import and connect (#4323)
timescale/docs: migrate/import-data-using-console.md

---
title: Import data using Tiger Cloud Console
excerpt: 
products: [cloud]
keywords: [import]
---

import ImportPrerequisitesCloudNoConnection from "versionContent/_partials/_prereqs-cloud-no-connection.mdx";
import EarlyAccessGeneral from "versionContent/_partials/_early_access.mdx";

# Import data into your $SERVICE_SHORT using $CONSOLE_LONG

> You can import data into your $SERVICE_SHORT using $CONSOLE_LONG. This page explains how to import CSV, Parquet, and text files, from your local machine and from an S3 bucket.

Reviewer: ```suggestion
This page shows you how to import data in CSV, Parquet, and text files stored on a source machine or in an S3 bucket into your $SERVICE_SHORT using $CONSOLE.
```

Author: Too much in one sentence, becomes hard to read.

> <Tabs label="Import files using Tiger Cloud Console">

Reviewer: Can you add a persistKey please. Check the existing ones.

<Tab title="From a local machine">

$CONSOLE_LONG enables you to drag and drop files to upload from your local machine.

<EarlyAccessGeneral /> 

## Prerequisites

<ImportPrerequisitesCloudNoConnection />

> <Tabs label="Upload files from a local machine">

Reviewer: A different peristKey please

<Tab title="From CSV">

To upload a CSV file to your $SERVICE_SHORT:

<Procedure>

> 1. **Select your $SERVICE_SHORT in $CONSOLE_SHORT, then click `Actions` > `Import data` > `Upload .CSV`**

Reviewer: Maybe a link to the services tab in console.

   ![Import from CSV into Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-import-csv.png)

> 1. **Click to browse, or drag and drop the file to import**

Reviewer: ```suggestion
1. **Click to browse, or drag the file to import**
```

1. **Configure the import**

   ![Configure the CSV import in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-configure-csv-import.png)

   - Set a delimiter.
   - Toggle to skip or keep the header.
   - Select to ingest the data into an existing table or create a new one. 
   - Provide the new or existing table name. 

>    - <Optional /> For a new table with a time column, toggle the time column to create a hypertable instead of a regular table.

Reviewer: $HYPERTABLE

1. **Click `Process CSV file`**

>    When the processing is finished, find your data imported to a new or existing table in `Explorer`. 

Reviewer: Something like: When processing is completed, to find the data your imported, click `Explorer`.

</Procedure>

</Tab>

<Tab title="From Parquet">

To upload a Parquet file to your $SERVICE_SHORT:

<Procedure>

> 1. **Select your $SERVICE_SHORT in $CONSOLE_SHORT, then click `Actions` > `Import data` > `Upload Parquet`**

Reviewer: Maybe a link to the services tab in console.

   ![Import from Parquet into Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-import-parquet.png)

> 1. **Click to browse, or drag and drop the file to import**

Reviewer: ```suggestion
1. **Click to browse, or drag the file to import**
```

1. **Configure the import**

   ![Configure the Parquet import in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-configure-parquet-import.png)

   - Select to ingest the data into an existing table or create a new one.
   - Provide the new or existing table name.

>    - <Optional /> For a new table with a time column, toggle the time column to create a hypertable instead of a regular table.

Reviewer: $HYPERTABLE

   
1. **Click `Process Parquet file`**

   When the processing is finished, find your data imported to a new or existing table in `Explorer`.

</Procedure>

</Tab>

<Tab title="From a text file">

To upload a TXT or MD file to your $SERVICE_SHORT:

<Procedure>

1. **Select your $SERVICE_SHORT in $CONSOLE_SHORT, then click `Actions` > `Import data` > `Upload text files`**

   ![Import from a text file into Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-import-txt.png)

1. **Click to browse, or drag and drop the file to import**
1. **Configure the import**

   Provide a name to create a new table, or select an existing table to add data to. 

   ![Configure the text file import in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-configure-txt-import.png)

1. **Click `Upload files`**

>    When the upload is finished, find your data imported to a new or existing table in `Explorer`.

Reviewer: Something like: When processing is completed, to find the data your imported, click Explorer.

</Procedure>

</Tab>

</Tabs>
    
</Tab>

<Tab title="From S3">

> $CONSOLE_LONG enables you to upload CSV and Parquet files, including compressed formats (GZIP and ZIP), by connecting to an S3 bucket.

Reviewer: ```suggestion
$CONSOLE_LONG enables you to upload CSV and Parquet files, including archives compressed using GZIP and ZIP, by connecting to an S3 bucket.
```

## Prerequisites

<ImportPrerequisitesCloudNoConnection />

- Ensure access to a standard Amazon S3 bucket containing your data files.
- Configure access credentials for the S3 bucket. The following credentials are supported:
   - [IAM Role][credentials-iam].
   - [Public anonymous user][credentials-public].

> <Tabs label="Import files from S3">

Reviewer: persist key please

<Tab title="Import from CSV">

    
To import a CSV file from an S3 bucket:

<Procedure>

1. **Select your $SERVICE_SHORT in $CONSOLE_SHORT, then click `Actions` > `Import from S3`**

> 1. **Point to your file in the S3 bucket**

Reviewer: ```suggestion
1. **Select your file in the S3 bucket**
```

   ![Import CSV from S3 in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-import-csv-from-s3.png)

   1. Provide your file path.

>    1. Select `CSV` in the file type dropdown.

Reviewer: ```suggestion
   1. Select `CSV`.
```

>    1. Select the authentication method: `IAM role` or `Public`. For `IAM role`, provide the role.

Reviewer: Make the options into a bulleted list please

   1. Click `Continue`.

1. **Configure the import**

   ![Configure CSV import from S3 in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-configure-csv-import-from-s3.png)

   - Set a delimiter.
   - Toggle to skip or keep the header.
   - Select to ingest the data into an existing table or create a new one.
   - Provide the new or existing table name.
   - <Optional /> For a new table with a time column, toggle the time column to create a hypertable instead of a regular table.

1. **Click `Process CSV file`**

>    When the processing is finished, find your data imported to a new or existing table in `Explorer`.

Reviewer: Something like: When processing is completed, to find the data your imported, click Explorer.

</Procedure>
    
</Tab>

<Tab title="Import from Parquet">

To import a Parquet file from an S3 bucket:

<Procedure>

1. **Select your $SERVICE_SHORT in $CONSOLE_SHORT, then click `Actions` > `Import from S3`**

> 1. **Point to your file in the S3 bucket**

Reviewer: ```suggestion
1. **Select your file in the S3 bucket**
```

   ![Import Parquet from S3 in Tiger Cloud](https://assets.timescale.com/docs/images/tiger-cloud-console/tiger-cloud-import-parquet-from-s3.png)

   1. Provide your file path.

>    1. Select `Parquet` in the file type dropdown.

Reviewer: ```suggestion
   1. Select `Parquet`.
```

>    1. Select the authentication method: `IAM role` or `Public`. For `IAM role`, provide the role.

Reviewer: List items please

   1. Click `Continue`.

1. **Configure the import**

>    - Select to ingest the data into an existing table or create a new one.

Reviewer: Make this the `text in the UI` please.

   - Provide the new or existing table name.

>    - <Optional /> For a new table with a time column, toggle the time column to create a hypertable instead of a regular table.

Reviewer: $HYPERTABLE

1. **Click `Process Parquet file`**

>    When the processing is finished, find your data imported to a new or existing table in `Explorer`.

Reviewer: Something like: When processing is completed, to find the data your imported, click Explorer.
